src/core/Entry.cpp

```diff
@@ -355,11 +367,20 @@ QString Entry::totpSeed() const
     m_data.totpStep = QTotp::defaultStep;
 
     if (m_attributes->hasKey("TOTP Settings")) {
-        QRegExp rx("(\\d+);(\\d)", Qt::CaseInsensitive, QRegExp::RegExp);
+        QString digitsRx = "(?:\\d+)";
+        // build a regex that supports any encoders that we support.
+        foreach (QTotp::Encoder encoder, QTotp::encoders) {
```

Comment:
Don't use the foreach macro, use C++11 range-based for loops (remember to cast to const to avoid detaching).